assets/components/MainMenu.svelte

```diff
@@ -20,6 +20,8 @@ let navEl;
 let searchHasFocus = false;
 let visibleLinks = [];
 
+let collapsedStates = JSON.parse(localStorage.getItem('collapsedStates')) || {};
```

Comment:
What happens to JSON.parse() when collapsedStates is not defined in localStorage?

Reply:
I think the || {} fallback should take care of that. If collapsedStates is not defined in localStorage, then it will be initialized as an empty object.

Reply:
I just tried it myself, and it seems like you're right. At least in Firefox. I'm not sure why though, since it doesn't look like parse() should be able to take null. Please change it just to be sure:

```javascript
let collapsedStates = JSON.parse(localStorage.getItem('collapsedStates') || '{}');
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse

Reply:
I'm not entirely sure what the difference is tbh, but from what I can tell it also doesn't hurt, so I implemented your snippet.